packages/mendel-pipeline/src/cache/index.js

```diff
@@ -1,29 +1,29 @@
 const path = require('path');
 const Entry = require('./entry.js');
+const variationMatches = require('mendel-development/variation-matches');
 
 class MendelCache {
     constructor(config) {
         this._store = new Map();
         this._baseConfig = config.baseConfig;
-        this._variationConfig = config.variationConfig;
-
-        const variationDirSet = new Set();
-        Object.keys(this._variationConfig.variations)
-            .filter(varKey => this._variationConfig.variations[varKey])
-            .forEach(varKey => {
-                this._variationConfig.variations[varKey].forEach(varFolderName => variationDirSet.add(varFolderName));
-            });
-        const varDirNames = Array.from(variationDirSet.keys());
-
-        this.variationalRegex = new RegExp(`(${varDirNames.map(dirName => `(${dirName})${path.sep}\\w+`).join('|')}|${this._baseConfig.dir})${path.sep}?`);
+        this._variations = config.variationConfig.variations;
     }
 
     getNormalizedId(id) {
-        if (isNodeModule(id)) return id;
-
-        // This is wrong WIP
-        // return id.replace(this.variationalRegex, '');
-        return id;
+        let normalizedId = id;
+
+        const match = variationMatches(this._variations, id);
```

Comment:
Not yet. But I don't think it will be part of this method to validate that, that method may be final and all the validation when the entry is being created. Will get to see when implementing such validations. The current implementation only covers the large portion of use cases, but anything that needs the value inside the `package.json` is incorrect so far.